sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_authentication_policy.py

```diff
@@ -46,13 +47,30 @@ async def send(self, request):
 
         await self.on_request(request)
 
+        original = None
+        # avoid aiohttp client closing the stream after sending the request
+        if request.http_request.body and hasattr(request.http_request.body, 'close'):
+            original = request.http_request.body.close
+            request.http_request.body.close = lambda: None
+
         response = await self.next.send(request)
 
         if response.http_response.status_code == 401:
             challenge = response.http_response.headers.get("WWW-Authenticate")
             if challenge and await self.on_challenge(request, response, challenge):
+                if request.http_request.body and hasattr(request.http_request.body, 'read'):
+                    try:
+                        # attempt to rewind the body to the initial position
+                        request.http_request.body.seek(0, SEEK_SET)
+                    except (UnsupportedOperation, ValueError, AttributeError):
```

Comment:
Hasn't this been built into the Core policy now?

Reply:
Nope.